.github/workflows/build-ironic-images.yaml

```diff
@@ -15,35 +15,61 @@ on:
     types: [checks_requested]
 
 jobs:
-  build-ironic-images:
+  build:
     runs-on: ubuntu-latest
-    defaults:
-      run:
-        working-directory: ironic-images/ipa-debian-bookworm
+    strategy:
+      matrix:
+        image:
+          - name: "ubuntu"
+            script: "keekz-ubuntu-extras"
+          - name: "debian"
+            script: "ipa-debian-bookworm"
     steps:
-      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4
-      - uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # v5
+      - name: Checkout repository
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4
+
+      - name: Setup python environment
+        uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # v5
         id: setup-python
         with:
           python-version: '3.11'
           cache: 'pip'
-      - run: sudo apt update && sudo apt install -y debootstrap qemu-utils
-      - run: pip install -r requirements.txt
+
+      - name: Install Dependencies
+        run: sudo apt update && sudo apt install -y debootstrap qemu-utils squashfs-tools kpartx
+
+      - name: Install python packages
+        run: pip install -r requirements.txt
         working-directory: ironic-images
-      - name: Build the IPA image
-        run: bash ipa-debian-bookworm.sh
+
+      - name: Build Images
+        run: |
+          bash "${{ matrix.image.script }}".sh
+          if [ "${{ matrix.image.name }}" == "ubuntu" ]; then
+            echo "qcow2_file=ironic-images/${{ matrix.image.script }}/${{ matrix.image.script }}.qcow2" >> $GITHUB_ENV
+          elif [ "${{ matrix.image.name }}" == "debian" ]; then
+            echo "initramfs_file=ironic-images/${{ matrix.image.script }}/${{ matrix.image.script }}.initramfs" >> $GITHUB_ENV
+            echo "kernel_file=ironic-images/${{ matrix.image.script }}/${{ matrix.image.script }}.kernel" >> $GITHUB_ENV
+          fi
+        working-directory: "ironic-images/${{ matrix.image.script }}"
         env:
           ELEMENTS_PATH: ${{ env.pythonLocation }}/share/ironic-python-agent-builder/dib:${{ github.workspace }}/ironic-images/custom_elements
-      - name: Dynamically set timestamp environment variable
-        run: echo "TIMESTAMP=$(date +"%Y%m%d%H%M%S")" >> $GITHUB_ENV
-      - name: Publish IPA Release
+
+      - name: Get commit hash
+        run: echo "COMMIT_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+
+      - name: Set timestamp environment variable
+        run: echo "TIMESTAMP=$(git show --no-patch --no-notes --pretty='%cd' --date=format:'%Y%m%d%H%M%S' ${{ env.COMMIT_HASH }})" >>  $GITHUB_ENV
```

Comment:
You can combine these into 1 run cause you're not using the COMMIT_HASH variable anywhere else. Additionally you shouldn't have to get the revision this way since it's available from the `github` context as `github.sha` see: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs

If you're wanting to output information then you can look at: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions

Reply:
Thanks, I used the github.sha suggestion.